Feature/navbar - edited the navbar component to match new designs in figma

ZhaoYuLL opened the pull request:
Closes #136 

- [x] Edited React Component
   - Edited the previous text blocks in the navbar
      - `"About"` -> `"Mission"`
      - `"Apply"` -> `"Get Involved"`
      - The actual links have not been changed
- [x] Mobile Responsive (hamburger button)
   - Renders the button using SVG
   - Used Tailwind to create a Desktop navigation menu and a mobile/small screen navigation menu
- [x] Buttons Link to respective pages
   - Created a navigation link array on the top of the file for easier changing of texts and links in the future

# Reference
![Image](https://github.com/user-attachments/assets/b0ec48c8-e119-4a33-aae9-19542503b630)

# Desktop View
<img width="1308" alt="image" src="https://github.com/user-attachments/assets/37f81948-5ff3-4f97-9d69-88b9ff6ed9c4">

# Mobile/Small Screen View (Expanded Dropdown Menu)
<img width="482" alt="image" src="https://github.com/user-attachments/assets/cbaf1971-c676-4170-a491-70108a654c18">

# Mobile/Small Screen View (Collapsed Dropdown Menu)
<img width="453" alt="image" src="https://github.com/user-attachments/assets/997326fc-21f9-4b0f-8dbd-eaa6df00cd20">


- [Figma Mockup](https://www.figma.com/design/Zw1lNlwMEahJCcN5X2L0WT/Blueprint-Website-Redesign-2024?node-id=1-2)

ZhaoYuLL committed:
formatted navbar to be more like figma design
- currently directly setting html to achieve hamburger effect for widths < 768px

ZhaoYuLL committed:
tailwind for hamburger icon

ZhaoYuLL committed:
adjusted dropdown menu

ZhaoYuLL committed:
ran prettier formatter

Caposto reviewed (changes requested):
Changes look great so far! Are you able to add some sort of bolding effect to the navbar links when a user hovers over them: I think adding ```hover:font-bold``` would do the trick.

Also, are you able to raise the font size and have the logo and links align more in the center like on the current iteration: https://sitblueprint.com/

Caposto reviewed (changes requested):
Also change '/about' to '/mission' and '/apply' to '/community' we are going to merge @j-ferber's PR first probably so that you will have the updated links. Reference: #141

ZhaoYuLL committed:
-added bolding effect for when user hovers over a link
-changed navbar appearance
-changed navbar links

ZhaoYuLL committed:
prettier formated

ZhaoYuLL commented:
yo made some changes:
-change '/about' to '/mission' and '/apply' to '/community'
-added bolding and a transition underline effection when user hovers over text
   - pure white bolding is really hard to see on the background so i used very light yellow
-changed appearnce of navbar to be like the current site
   - Same padding as before
   
# Navbar changes (Left browser has current running site, right is current branch)
   
<img width="1371" alt="image" src="https://github.com/user-attachments/assets/7bdb45af-3b7e-4578-bd01-006ee04bd7f7">
<img width="1363" alt="image" src="https://github.com/user-attachments/assets/df398723-dfd1-4303-a7e6-19170fda7ae3">

# Hovering on Desktop Navbar
<img width="1073" alt="image" src="https://github.com/user-attachments/assets/4a2f1d44-a4db-47b3-a163-aa81486c376f">

# Hovering on Mobile Dropdown Menu
<img width="836" alt="image" src="https://github.com/user-attachments/assets/f2b24d23-084a-46ac-8c0d-e3ebda3e9552">

j-ferber committed:
Merge branch 'main' into feature/navbar

j-ferber committed:
Fixed formatting after merge

miguel-merlin commented:
@j-ferber Usually on feature branches its better to rebase. Next time you can try
`git pull origin main --rebase`

j-ferber commented:
@miguel-merlin my bad, will do next time. did it affect the PR?

miguel-merlin commented on src/_components/Navbar.jsx:
```diff
@@ -1,29 +1,109 @@
-        <img src={logo} alt="Logo" className="mr-2 w-10 h-10" />
-        <span className="self-center text-3xl font-semibold font-poppins whitespace-nowrap dark:text-white">
+    <nav className="relative flex items-center justify-between p-6 bg-primary z-50">
+      {/* Logo and Brand Name Section */}
```
I don't know to what extent all the comments in the return statement make the code verbose. What do you think @Caposto @ShawnAviles?

ShawnAviles replied:
I agree, it is a little verbose. Since this is mostly JSX and not really any complicated logic, I don't think the comments are necessary. You had the right idea @ZhaoYuLL, but I think your code is self-explanatory and readable as is

ZhaoYuLL replied:
I removed the verbose comments

miguel-merlin commented on src/_components/Navbar.jsx:
```diff
@@ -1,29 +1,109 @@
+
+// Function to render desktop navigation links
+function renderDesktopNavigationLinks() {
+  // Map through the links to render them
```
Same comment here on the unnecessary comments

ShawnAviles replied:
Agreed. This is a good example where a well named function doesn't need the extra comments (i.e. your function name is the same as the comment here)

ZhaoYuLL replied:
Removed

miguel-merlin commented on src/_components/Navbar.jsx:
```diff
@@ -1,29 +1,109 @@
+        {/* Mobile Navigation Menu - Expands and shows when hamburger icon is clicked*/}
+
+        {/* Change background color for dropdwon menu here if desired */}
+        {/* current dropdown menu css:
```
@sahanaali What do you think of the background color of the navbar?

miguel-merlin commented on src/_components/Navbar.jsx:
```diff
@@ -1,29 +1,109 @@
+      {/* Mobile Menu Toggle and Dropdown - Visible on screens smaller than "large" */}
+      <div className="lg:hidden relative">
+        {/* Toggles the mobile menu */}
+        <input
```
Could you instead use a `useState` hook here. I don't think it is the best approach to use an input checkbox to toggle the menu with the links. You could maybe do something like
```
const [isMobileMenuOpen, setIsMobileMenuOpen] = useState(false);
{/* Code */}
        <button
          onClick={() => setIsMobileMenuOpen(!isMobileMenuOpen)}
          className="block cursor-pointer p-2"
          aria-label="Toggle mobile menu"
          aria-expanded={isMobileMenuOpen}
        >
          {renderHamburgerIcon()}
        </button>
```

ShawnAviles replied:
Using this checkbox approach is mainly concerning for accessibility reasons. Per our talk @ZhaoYuLL, I know i mentioned staying away from using the react state for the tracking window size changes as it was weird, but I think this true/false state toggle is a valid approach. My fault on the misdirection tho 

If anyone is interested, here are some discussions on this "checkbox hack" topic: [some dev article](https://justmarkup.com/articles/2019-12-04-hamburger-menu/#the-checkbox-hack) and [a related reddit post](https://www.reddit.com/r/webdev/comments/10gpdxn/the_checkbox_hack_pros_and_cons/).

miguel-merlin replied:
Didn't know about this. You can leave it as is.

ZhaoYuLL replied:
Ok I'll leave it as is for now

miguel-merlin reviewed (changes requested):
Mega job! Please review the comments I have made.

miguel-merlin commented:
Nah, you're good @j-ferber

ShawnAviles commented on src/_components/Navbar.jsx:
```diff
@@ -1,29 +1,109 @@
+        <img src={logo} alt="Blueprint Logo" className="mr-2 w-10 h-10" />
+        <span className="text-3xl font-poppins font-semibold hover:font-bold">
           blueprint
         </span>
```
Can you change the navbar and buttons to be larger on mobile? Mobile users are using their thumb instead of a cursor so usability is something to keep in mind 😃 

Something like this
```
<nav className="relative flex items-center justify-between lg:p-6 p-10 bg-primary z-50">
  {/* Logo and Brand Name Section */}
  <a href="/" className="flex items-center text-white">
  <img src={logo} alt="Blueprint Logo" className="mr-2 lg:w-10 lg:h-10 w-16 h-16" />
  <span className="lg:text-3xl text-5xl font-poppins font-semibold hover:font-bold">
    blueprint
  </span>
  </a>
    ...
```

**Note**: you'd also have to increase the size of the text in the dropdown and the hamburger menu itself (`w-6` to `w-10`) too

ShawnAviles reviewed (changes requested):
nice job, just some slight changes. sorry for the barrage of comments

ZhaoYuLL committed:
removed verbose comments

ZhaoYuLL committed:
increased size of hamburger icon and dropdown menu

ZhaoYuLL committed:
reformatted for prettier

ZhaoYuLL commented:
Removed verbose comments and increased dropdown menu size, text size, etc, as well as the hamburger icon. Let me know if you want me to change the size again

# Mobile Display
![image](https://github.com/user-attachments/assets/2eeb344c-1508-4f5b-8c90-e6ec9f74414d)

miguel-merlin reviewed (approved):
lgtm!

ShawnAviles reviewed (approved):
Looks great. Nice job! 🥶
